components/mpas-ocean/src/shared/mpas_ocn_frazil_forcing.F

```diff
@@ -600,8 +606,8 @@ subroutine ocn_frazil_forcing_build_arrays(domain, meshPool, forcingPool, stateP
                                                  / dt
```

Comment:
Have you considered including the advective flux of water at the freezing temperature? I believe this assumes the freezing temperature is 0.

Reply:
The temperature tendency term would have two parts, the latent heat term that you have here as well as the temperature change due to mixing water of a different temperature. Based on my reading of the code, we have neglected the latter, which essentially means an assumption of 0 degC for the melted frazil water. As an example, @irenavankova allowed the option to have subglacial water enter at the freezing point https://github.com/E3SM-Project/E3SM/blob/de2142e485ebf90d627965f725ee226686b36c6d/components/mpas-ocean/src/shared/mpas_ocn_surface_bulk_forcing.F#L843-L846

Reply:
the brine has the same temperature as the ice.

Reply:
I think @cbegeman is right and the expelled brine should cool the top layer in addition to salinifying it.  @njeffery, are you arguing that this is already accounted for in the surface freezing term because the brine is at the freezing temperature?

Reply:
To try to clarify, it seems that right now we have a default freezing temperature in the open ocean of 0 degC (with no salinity or pressure dependence) so the assumption that the water enters at 0 degC is correct and we don't need to add any additional term for the mixing of the melted water at the freezing temperature with the surrounding ocean water. It's only in ice shelves that this assumption is incorrect because we use a different freezing temperature. Not at all saying that this is something this PR needs to address; I brought it up because I was wondering if it was something that you had considered or tested. Seems like this is something to explore separately.

Reply:
@cbegeman : The freezing temperature in the open ocean is not 0 deg C.  It's a function of salinity and pressure.

Reply:
Good to know. I was misled by these lines: https://github.com/E3SM-Project/E3SM/blob/b6e16c66f9983813ffe6b91648c8dc8f3d1903f9/components/mpas-ocean/bld/namelist_files/namelist_defaults_mpaso.xml#L543-L546

In that case, the assumption that the melted frazil water is at 0 degC, implicit in neglecting this advective (non-latent) d(hT)/dt term, is false in both open ocean and ISC. This does predate this PR so we can address it elsewhere.

Reply:
@cbegeman, I don't think frazil is allowed to melt in the ocean anywhere so I don't think any assumption is currently made about its temperature.  In the open ocean, frazil gets passed to the sea ice model, which could melt it.  In ice-shelf cavities, there is no mechanism for melting frazil at all at the moment.  It could be passed to the ice-sheet model in the future, and then be allowed to melt at the interface.

Reply:
> I don't think frazil is allowed to melt in the ocean anywhere 

I will amend that, since it is possible for frazil to form deeper in the ocean and then melt at intermediate depths (in which case I believe the local freezing point is correctly accounted for).  But there is no mechanism for it to melt at the surface within the ocean model.

Reply:
Yes, what I'm thinking of is the point intermediate in the water column where some of the frazil ice mass that has been accumulated from deeper melts. Suppose the water in that layer is 2 degC and the freezing point is -1 degC. The way we have it written right now, we extract the latent heat for melting but we neglect the term that would mix that -1 degC melted water with the 2 degC water. By assuming that term is 0, it's like assuming that melted frazil water enters at a temperature of 0 degC.

Reply:
Right, and with this porous approach, don't we make that assumption especially strongly at the surface, where we dump all the brine that that the frazil is carrying?

Reply:
It feels to me like we maybe need one ore more design docs for frazil on confluence (like https://acme-climate.atlassian.net/wiki/spaces/PSC/pages/4124966913/Antarctic+runoff+in+Polar+configurations+-+data+iceberg+and+ice-shelf+melt+forcing) where we can discuss both Nicole's implementation and what's currently missing (regardless of porosity) in terms of heat fluxes.

@njeffery, would you be up for describing what you've done in a design doc if you haven't already?

Reply:
Yes.   That's on my to-do list.  I'll let you know when it's written.